### ISSUE TITLE
Refactor: unify AJAX controllers on a shared base (guard, DI, registration)

### ISSUE THREAD

**Issue by rpnunez** [rpnunez/wp-ai-scheduler#1943]
## Context

PR #1934 unified a family of "similar" classes that each did the same plumbing in inconsistent ways. The AJAX controller family shows the same signature at much larger scale. There are 26 `ai-post-scheduler/includes/class-aips-*-controller.php` classes with **no shared base** — the only `extends` is `AIPS_Admin_Flow_Controller extends AIPS_Campaigns_Controller` (`class-aips-admin-flow-controller.php:13`), itself a smell.

The **response layer is already unified** (786 `AIPS_Ajax_Response` calls, zero raw `wp_send_json_*` left in any controller), so this refactor is scoped to the *guard + DI + registration* half only.

## Problem

- **Copy-pasted guard.** The 6-line nonce+capability preamble is inlined ~140× (e.g. `class-aips-authors-controller.php:95`; 18× in `class-aips-schedule-controller.php`, 20× in `class-aips-author-topics-controller.php`, 13× in `class-aips-internal-links-controller.php`):
  ```php
  if ( ! check_ajax_referer('aips_ajax_nonce', 'nonce', false) ) {
      AIPS_Ajax_Response::error(__('Invalid nonce.', 'ai-post-scheduler'));
  }
  if (!current_user_can('manage_options')) {
      AIPS_Ajax_Response::permission_denied();
  }
  ```
- **Re-invented under 4+ names** (proof the base is missing): `authorize()` `class-aips-post-slices-controller.php:43`; `ajax_guard()` `class-aips-campaigns-controller.php:619`; `verify_request()` in `class-aips-affiliate-links-controller.php:213`, `class-aips-system-status-controller.php:70`, `class-aips-stress-test-controller.php:148`, `class-aips-settings-ajax.php:165`; `verify_nonce_and_cap()` `class-aips-cache-monitor-controller.php:409`.
- **Latent bug.** `AIPS_Campaigns_Controller::ajax_ai_generate_campaign()` `:580` returns `AIPS_Ajax_Response::permission_denied()` for a **nonce** miss (wrong code/status) — and doesn't even use its own `ajax_guard()` helper.
- **Inconsistency.** Nonce action names diverge (`aips_ajax_nonce` vs per-endpoint `aips_source_data_*`, `aips_get_telemetry*`, `aips_campaign_wizard_ai_generate`); capability constants diverge (`manage_options` vs `edit_posts`/`edit_post` in `class-aips-ai-edit-controller.php`, `unfiltered_html` in sources); failure response shapes diverge.
- **Mixed DI.** Container-with-fallback in newer controllers (telemetry `:29`, system-status `:40`, campaigns `:63`) vs. 6 hard-`new`s in `class-aips-authors-controller.php:73` and `::instance()` singletons elsewhere. `AIPS_Container::makeIfExists($id, $fallback)` (`class-aips-container.php:177`) already exists for the uniform path.
- **Double source of truth for registration.** Each action lives in both `AIPS_Ajax_Registry::$map` and the controller constructor's `add_action()`.

## Plan

1. **Introduce `abstract AIPS_Ajax_Controller_Base`** (or a `Trait_AIPS_Ajax_Guard` if inheritance is undesirable) exposing one canonical guard:
   ```php
   protected function verify_request(
       string $nonce = 'aips_ajax_nonce',
       string $capability = 'manage_options'
   ): void
   ```
   which emits a single canonical failure shape (nonce miss → `error('Invalid nonce.')`; capability miss → `permission_denied()`), fixing the campaigns semantics bug.
2. **Make the base constructor container-aware** — resolve collaborators via `AIPS_Container::makeIfExists()` with a `new` fallback, matching the DI pattern #1934 applied to `AIPS_Author_Post_Generator`. Controllers that hard-`new` (authors et al.) migrate to injected/resolved deps.
3. **Migrate controllers one at a time.** Replace the ~140 inline blocks and the 6 bespoke helpers with `verify_request(...)`; pass a per-action nonce/capability where the endpoint legitimately differs (ai-edit's `edit_post`, sources' `unfiltered_html`, telemetry/sources per-action nonces). Each controller is an independent, low-risk PR.
4. **(Optional, follow-up)** Let the base own registration by iterating an `$actions` map so the registry entry and `add_action()` stop being two sources of truth.

## Files touched

- New: `includes/class-aips-ajax-controller-base.php` (or `includes/trait-aips-ajax-guard.php`)
- Each `includes/class-aips-*-controller.php` (migrated per-slice), starting with the ones that already reinvented the guard (post-slices, campaigns, affiliate-links, system-status, stress-test, cache-monitor) plus `class-aips-settings-ajax.php`.

## Reuse

`AIPS_Ajax_Response`, `AIPS_Ajax_Registry`, `AIPS_Container` (`makeIfExists`), and the existing central dispatcher `boot_ajax()` (`ai-post-scheduler.php`).

## Verification

- `php -l` on every touched file; `composer lint:repository-boundary`.
- PHPUnit with `AIPS_WP_TEST_SKIP_DB_CREATE=true`: a base-guard test asserting nonce-miss vs. capability-miss produce the canonical (and now correct) response shapes; the campaigns handler returns an `error` shape (not `permission_denied`) on a nonce miss.
- Manual smoke: one representative AJAX endpoint per migrated controller (save/delete/list) to confirm behavior is unchanged.

## Notes / risk

Large but highly mechanical; low risk **per slice**. Do NOT attempt in one PR — migrate per controller. Guarded by the `ajax-controller-reviewer` subagent / `ajax-controller-changes` skill (nonce / capability / sanitization / registry compliance).
